vault/resource_aws_secret_backend_role.go

```diff
@@ -128,6 +145,9 @@ func awsSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 	if d.HasChange("iam_groups") {
 		data["iam_groups"] = iamGroups
 	}
+	if d.HasChange("user_path") {
```

Comment:
It might be a good idea to log a warning when the user path is configured and the `credential_type` is not `iam_user`.